lib/rubygems/server.rb

```diff
@@ -81,20 +82,20 @@ class Gem::Server
     <b><%=h spec["name"]%> <%=h spec["version"]%></b>
 
     <% if spec["ri_installed"] || spec["rdoc_installed"] then %>
-      <a href="<%=u spec["doc_path"]%>">[rdoc]</a>
+      <a href="<%=spec["doc_path"]%>">[rdoc]</a>
```

Comment:
No, it's already sanitized in the doc_root method which generates doc_path. doc_root calls ERB::Util.u on gem_name, which is the only externally provided component. Using uri_encode here would be redundant, and a less thorough option in place of sanitizing in doc_root. (uri_encode is only necessary when sanitizing a full URL address where functional URL components, ie '://', should not be encoded. ERB::Util.u is a more thorough encoding method that should be used when strings are inserted into a URL and functional URL components, ie '://', should be encoded)